ceph.spec.in

```diff
@@ -813,18 +816,42 @@ export CPPFLAGS="$java_inc"
 export CFLAGS="$RPM_OPT_FLAGS"
 export CXXFLAGS="$RPM_OPT_FLAGS"
 
-env | sort
-
+# Parallel build settings ...
+# unlimit _smp_mflags in system macro
+%global _smp_ncpus_max 0
+CEPH_MFLAGS_JOBS="%{?_smp_mflags}"
 %if %{with lowmem_builder}
-%if 0%{?jobs} > 8
-%define _smp_mflags -j8
-%endif
+lo_jobs="%{?jobs:%{jobs}}"
+%if 0%{?suse_version}
+%ifarch %arm
+    %if ! 0%{?qemu_user_space_build:1}
+        # native hardware is not that powerful
+        lo_jobs="1"
+    %endif
+%endif
+%endif
+# do not eat all memory
+echo "Available memory:"
+free
+echo "System limits:"
+ulimit -a
+lo_jobs_reduced=
+if test -n "$lo_jobs" -a "$lo_jobs" -gt 1 ; then
+    mem_per_process=1600
+    max_mem=$(LANG=C free -t -m | sed -n "s|^Mem: *\([0-9]*\).*$|\1|p")
+    max_jobs="$(($max_mem / $mem_per_process))"
+    test "$lo_jobs" -gt "$max_jobs" && lo_jobs="$max_jobs" && lo_jobs_reduced="yes" && echo "Warning: Reducing build parallelism to -j$max_jobs because of memory limits"
+    test "$lo_jobs" -le 0 && lo_jobs=1 && echo "Warning: Not using parallel build at all because of memory limits"
+fi
+# run make
+if test -n "$lo_jobs" ; then
+  CEPH_MFLAGS_JOBS="-j$lo_jobs"
+fi
 %endif
-
-# unlimit _smp_mflags in system macro if not set above
-%define _smp_ncpus_max 0
-# extract the number of processors for use with cmake
-%define _smp_ncpus %(echo %{_smp_mflags} | sed 's/-j//')
+export CEPH_MFLAGS_JOBS
+export CEPH_SMP_NCPUS=$(echo "$CEPH_MFLAGS_JOBS" | sed 's/-j//')
+ 
+env | sort
```

Comment:
All of this seems way too complicated. Since we are actually computing the memory why don't we just do this for _any_ builder, not just lowmem builders? In that case, we could simply directly use the lo_jobs macro later and not dance around it like we do, now :-( .

Reply:
@b-ranto By doing it this way, one is not exposed to the memory computation and using of the calculation result to set the `-j` value, unless you have the "lowmem_builder" bcond set.

It didn't occur to me at the time that this might be appealing outside of the OBS.

Reply:
@smithfarm I noticed that the memory computation takes place for lowmem_builder only but it might help if we just computed it all the time as there are all sorts of machines that do not have enough memory for the number of cpus/threads they have (this is especially true if you take into consideration things like hyper-threading).

We do occassionally hit a build error even with koji/brew because we run out of memory. It would also make the code much more clear/easier to read.

Reply:
@b-ranto Yup. I'll get rid of lowmem_builder bcond and repush.